isis/src/control/apps/findfeatures/FastGeom.cpp

```diff
@@ -62,24 +67,398 @@ FastGeom::~FastGeom() { }
+        r_angle += d_theta;
+      }
+
+      // Update radial distance from image center and points/circle
+      fg_radius         += fg_radius_inc;
+      points_per_radius += fg_point_inc;
+    }
+  }
+  else { // true == grid_method
+
+#if USE_OLD_ALGORITHM
+    mapper = train.source().getGeometryMapping(query.source(),
+                                               m_fastpts,
+                                               m_tolerance);
+#endif
+
+    logit << "--> Using Grid Algorithm train-to-query mapping <--\n";
+
+     // Compute increment
+    int   fg_minpts = qMax(m_fastpts, 16);
+    int increment = (int) (std::sqrt(std::max(24.0, (double) (fg_minpts - 1))) + 1.0);
+
+    double fg_max_axis = qMax( qMax(q_max_samp, q_max_line), qMax(t_max_samp, t_max_line) );
+    int v_max_iter     = int( fg_max_axis / 2.0 );
+
+    int fg_grid_start_iter = toInt(  m_parameters.get("FastGeomGridStartIteration", "0") );
+    int fg_grid_stop_iter  = toInt(  m_parameters.get("FastGeomGridStopIteration", toString(v_max_iter)) );
+    int fg_grid_iter_step  = toInt(  m_parameters.get("FastGeomGridIterationStep", "1") );
+    bool fg_save_all       = toBool( m_parameters.get("FastGeomGridSaveAllPoints", "false") );
+
+    logit <<   "\n  FastGeomGridStartIteration: " << fg_grid_start_iter << "\n";
+    logit <<   "  FastGeomGridStopIteration:  " << fg_grid_stop_iter << "\n";
+    logit <<   "  FastGeomGridIterationStep:  " << fg_grid_iter_step << "\n";
+    logit <<   "  FastGeomGridSaveAllPoints:  " << toString(fg_save_all) << "\n";
+    logit <<   "  FGPointIncrement:           " << increment << "\n";
+
+
+    bool done = false;
+    int iteration = fg_grid_start_iter;
+    int currinc(0);
+    double v_q_nsamps = q_max_samp - q_min_samp + 1.0;
+    double v_q_nlines = q_max_line - q_min_line + 1.0;
+
+    int n_iterations(0);
+    while ( (iteration < fg_grid_stop_iter) && ( !done) ) {
+
+      currinc = increment + ( iteration * 2 );
+      n_iterations++;
+
+      double sSpacing = qMax( 1.0, v_q_nsamps/(currinc*1.0) );
+      double lSpacing = qMax( 1.0, v_q_nlines/(currinc*1.0) );
+
+      if ( qMax( sSpacing, lSpacing ) <= 1.0 ) done = true; // Last possible loop
+
+      // To get all points computed...
+      if ( !fg_save_all ) {
+        q_points.clear();
+        t_points.clear();
+        t_inFOV.clear();
+        q_surface_points.clear();
+      }
+
+      // Clear these for the matrix transform
+      q_infov_points.clear();
+      t_infov_points.clear();
+      n_train_fov = 0;
+
+      // Run the loop
+      for (int l = 0 ; l < currinc ; l++) {
+        for ( int s = 0 ; s < currinc ; s++ ) {
+
+          // Total point count
+          n_total_points++;
+
+          double q_y = (lSpacing / 2.0 + lSpacing * l + 0.5) + q_min_line;
+          double q_x = (sSpacing / 2.0 + sSpacing * s + 0.5) + q_min_samp;
+
+          // Check to see if the point is in the image FOV
+          FGPoint q_coord(q_x, q_y);
```

Comment:
Same as the ring algorithm, you may as well just define the point directly.

Reply:
Done.